### PR TITLE
[Workers AI] TOC revision + storage options name change in Platform sections

### PR DESCRIPTION
**Pull request by deadlypants1973**
PCX-10034

Two tasks in one PR:
1. All Dev Plat products "Storage options guide" has been changed to the new "Choose a storage or data product" title following Matt's PR.
2. WAI Platform section has been updated according to content strategy.

Regarding task 2, review the following resources to understand the purpose of this change before commenting:

PCX-9882
PCX Developer Platform Information Architecture (IA) in wiki
https://developers.cloudflare.com/style-guide/documentation-content-strategy/information-architecture/

### CODE REVIEW

File: content/workers-ai/platform/limits.md [hunk 1]
```diff
@@ -10,13 +10,13 @@ During the open beta, the following limits are in place:
 
 **Inference requests per minute (per model)**
 
-- [@cf/meta/llama-2-7b-chat-int8](/workers-ai/models/#text-generation) - 50 reqs/min
-- [@cf/openai/whisper](/workers-ai/models/#automatic-speech-recognition) - 4000 reqs/min
-- [@cf/meta/m2m100-1.2b](/workers-ai/models/#translation) - 4000 reqs/min
-- [@cf/huggingface/distilbert-sst-2-int8](/workers-ai/models/#text-classification) - 6000 reqs/min
-- [@cf/microsoft/resnet-50](/workers-ai/models/#image-classification) - 6000 reqs/min
-- [@cf/baai/bge-base-en-v1.5](/workers-ai/models/#text-embeddings) - 6000 reqs/min
+- [@cf/meta/llama-2-7b-chat-int8](/workers-ai/models/#text-generation) - 50 requests/minute.
```
Comment (craigsdennis) [inline content/workers-ai/platform/limits.md:13]: These have proper model pages now under /models

Comment (craigsdennis) [reply content/workers-ai/platform/limits.md:13]: And will match the last bit after the / in the description

**Review by craigsdennis — APPROVED** [review]
lgtm!

**Review by Maddy-Cloudflare — APPROVED** [review]

File: content/workers-ai/platform/limits.md (hunk 1 above)
Comment (deadlypants1973) [reply content/workers-ai/platform/limits.md:13]: will address this in individual page revisions!

**Pull request merged**